control/timeresp.py

```diff
@@ -319,26 +327,28 @@ def forced_response(sys, T=None, U=0., X0=0., transpose=False,
                 n_steps = U.shape[0]
             else:
                 n_steps = U.shape[1]
-            T = np.array(range(n_steps)) * (1 if sys.dt is True else sys.dt)
+            dt = 1. if sys.dt in [True, None] else sys.dt
+            T = np.array(range(n_steps)) * dt
         else:
             # Make sure the input vector and time vector have same length
             # TODO: allow interpolation of the input vector
             if (U.ndim == 1 and U.shape[0] != T.shape[0]) or \
                     (U.ndim > 1 and U.shape[1] != T.shape[0]):
                 ValueError('Pamameter ``T`` must have same elements as'
                            ' the number of columns in input array ``U``')
+    else:
+        if T is None:
+            raise ValueError('Parameter ``T`` is mandatory for continuous '
+                             'time systems.')
 
     # Test if T has shape (n,) or (1, n);
-    # T must be array-like and values must be increasing.
-    # The length of T determines the length of the input vector.
-    if T is None:
-        raise ValueError('Parameter ``T``: must be array-like, and contain '
-                         '(strictly monotonic) increasing numbers.')
     T = _check_convert_array(T, [('any',), (1, 'any')],
                              'Parameter ``T``: ', squeeze=True,
                              transpose=transpose)
+
+    # equally spaced also implies strictly monotonic increase
     dt = T[1] - T[0]
-    if not np.allclose(T[1:] - T[:-1], dt):
+    if not np.allclose(np.diff(T), np.full_like(T[:-1], dt)):
```

Comment:
This sees fine, but `np.allclose(np.diff[T], dt)` seems cleaner (`dt` will broadcast to the right shape).

Reply:
Thanks. I did not see a broadcast documented for `np.testing.assert_allclose` and assumed this might stop working in the future. But the broadcast is mentioned for `np.allclose`.